code/modules/power/lighting.dm

```diff
@@ -140,12 +141,21 @@
 	var/bulb_colour = COLOR_WHITE
 	var/status = LIGHT_OK		// LIGHT_OK, _EMPTY, _BURNED or _BROKEN
 	var/flickering = FALSE
+	///what's the duration that the light switches between on and off while flickering
+	var/flicker_time = 2 SECONDS
 	var/light_type = /obj/item/light_bulb/tube		// the type of light item
 	var/fitting = "tube"
 	///count of number of times switched on/off. this is used to calc the probability the light burns out
 	var/switchcount = 0
 	/// true if rigged to explode
 	var/rigged = FALSE
+	///holds the state of our flickering
+	var/light_flicker_state = FALSE
+	///if true randomize the time we turn on and off
+	var/random_flicker = FALSE
+	var/flicker_time_upper_max = 10 SECONDS
+	var/flicker_time_lower_min = 0.2 SECONDS
+	var/datum/looping_sound/flickeringambient/lightambient
```

Comment:
honestly this whole light flicker thing is probably better as a component.
Also autodoc everywhere.